src/ui/contexts/TransactionsContext.tsx

```diff
@@ -37,52 +38,59 @@ export function TransactionsContextProvider({
+    let addressOrPair: AddressOrPair = keyPair;
+    let signer = undefined;
+
+    // If the account is not a testing account (//Alice etc.),
+    // we need the `signer` from the extension
+    if (!keyPair.meta.isTesting) {
+      signer = (await web3FromAddress(accountId)).signer;
+      // Only use plain address, otherwise pjs api want to sign with given KeyPair
+      // instead of `signer`
+      addressOrPair = keyPair.address;
+    }
 
     try {
-      const unsub = await extrinsic.signAndSend(
-        account,
-        { signer: injector?.signer || undefined },
-        async result => {
-          if (result.isInBlock || result.isFinalized) {
-            const events: Record<string, number> = {};
-
-            result.events.forEach(record => {
-              const { event } = record;
-              const key = `${event.section}:${event.method}`;
-              if (!events[key]) {
-                events[key] = 1;
-              } else {
-                events[key]++;
-              }
-            });
-
-            if (!isValid(result)) {
-              setTxs({ ...txs, [id]: { ...tx, status: TxStatusMap.Error, events } });
-
-              let message = 'Transaction failed';
-
-              if (result.dispatchError?.isModule) {
-                const decoded = api?.registry.findMetaError(result.dispatchError.asModule);
-                message = `${decoded?.section.toUpperCase()}.${decoded?.method}: ${decoded?.docs}`;
-              }
-
-              onError && onError(result);
-
-              throw new Error(message);
+      const unsub = await extrinsic.signAndSend(addressOrPair, { signer }, async result => {
+        if (result.isInBlock || result.isFinalized) {
+          const events: Record<string, number> = {};
+
+          result.events.forEach(record => {
+            const { event } = record;
+            const key = `${event.section}:${event.method}`;
+            if (!events[key]) {
+              events[key] = 1;
+            } else {
+              events[key]++;
             }
+          });
+
+          if (!isValid(result)) {
+            setTxs({ ...txs, [id]: { ...tx, status: TxStatusMap.Error, events } });
 
-            onSuccess && (await onSuccess(result));
+            let message = 'Transaction failed';
 
-            setTxs({ ...txs, [id]: { ...tx, status: TxStatusMap.Success, events } });
+            if (result.dispatchError?.isModule) {
+              const decoded = api?.registry.findMetaError(result.dispatchError.asModule);
+              message = `${decoded?.section.toUpperCase()}.${decoded?.method}: ${decoded?.docs}`;
+            }
 
-            unsub();
+            onError && onError(result);
 
-            nextId++;
+            throw new Error(message);
           }
-        },
-      );
+
+          onSuccess && (await onSuccess(result));
+
+          setTxs({ ...txs, [id]: { ...tx, status: TxStatusMap.Success, events } });
+
+          unsub();
```

Comment:
Need a quick comment here.

Reply:
Made it a bit more clear in [my recent commit](https://github.com/paritytech/contracts-ui/pull/527/commits/8fd0d50c9490e74c31e88d958fd086a9e7b9faad). This part adds the current `tx` as a success to the `txs` map and `unsubscribes` from the submitted `extrinsic.singAndSend()` extrinsic status updates as the work is done here.

But a bit sad the code only seem to care about `finalized` and `inBlock` events. 🤷 

https://github.com/paritytech/contracts-ui/blob/8fd0d50c9490e74c31e88d958fd086a9e7b9faad/src/ui/contexts/TransactionsContext.tsx#L58